window.c

```diff
@@ -201,6 +203,24 @@ void win_open(win_t *win)
 	if (win->xwin == None)
 		error(EXIT_FAILURE, 0, "Error creating X window");
 
+  // set the _NET_WM_PID
+  pid_t pid = getpid();
+  XChangeProperty(e->dpy, win->xwin,
+                  atoms[ATOM__NET_WM_PID], XA_CARDINAL, sizeof(pid_t) * 8,
+                  PropModeReplace, (unsigned char *) &pid, 1);
+
+  // set the WM_CLIENT_MACHINE
+  char hostname[255];
+  if (gethostname(hostname, sizeof(hostname)) == 0) {
+      XTextProperty tp;
+      tp.value = (unsigned char *)hostname;
+      tp.nitems = strlen(hostname);
```

Comment:
`gethostname(2)` is not guaranteed to include a terminating null byte if the hostname is longer than the buffer, so it would be safer to use `strnlen(3)` here.

> If the null-terminated hostname is too large to fit,  then the  name is truncated, and no error is returned (but see NOTES below).  POSIX.1 says that if such truncation occurs, then it is unspecified whether the returned buffer includes a terminating null byte.

```suggestion
      tp.nitems = strnlen(hostname, sizeof(hostname));
```